pkg/apis/acme/v1alpha3/types_order.go

```diff
@@ -65,21 +65,15 @@ type OrderSpec struct {
 	IssuerRef cmmeta.ObjectReference `json:"issuerRef"`
 
 	// CommonName is the common name as specified on the DER encoded CSR.
-	// If CommonName is not specified, the first DNSName specified will be used
-	// as the CommonName.
-	// At least one of CommonName or a DNSNames must be set.
+	// If specified, this value must also be present in `dnsNames`.
```

Comment:
`dnsNames`. As I mentioned before, these docs are written for API consumers and not users of the Golang API (simply because there are more people using this via the API than there are Golang SDK).

This also follows what Kubernetes do upstream, and results in more readable `kubectl explain` output etc